components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/HostObjectUtils.java

```diff
@@ -100,6 +101,32 @@ protected static SubscriberKeyMgtClient getKeyManagementClient() throws APIManag
         }
     }
 
+    /**
+     * Used to get instance of ProviderKeyMgtClient
+     * @return ProviderKeyMgtClient
+     * @throws APIManagementException
+     */
+    protected static ProviderKeyMgtClient getProviderClient() throws APIManagementException {
+        APIManagerConfiguration config = HostObjectComponent.getAPIManagerConfiguration();
+        String url = config.getFirstProperty(APIConstants.API_KEY_MANAGER_URL);
+        if (url == null) {
+            handleException("API key manager URL unspecified");
+        }
+
+        String username = config.getFirstProperty(APIConstants.API_KEY_MANAGER_USERNAME);
+        String password = config.getFirstProperty(APIConstants.API_KEY_MANAGER_PASSWORD);
+        if (username == null || password == null) {
```

Comment:
This condition will not become true if the username or password is set to an empty string. Better to use StringUtils.isEmpty() method.